### PR TITLE
fix(test): making tests type-safe 

### LINKED ISSUE #299: [tests] Update bridge related tests to be type-safe and clean them up

### PR DESCRIPTION
**Pull request by RogerLamTd**

### CODE REVIEW

**Comment by RogerLamTd** [comment]
isMessageSent, isSignalReceived and isMessageReceived tests are done, waiting on @cyberhorsey to unblock retryMessage. Converted a bunch of tests to use type-chain for type safety. Please review! @dantaik @davidtaikocha

File: packages/protocol/contracts/libs/LibConstants.sol [hunk 1]
```diff
@@ -17,7 +17,6 @@ library LibConstants {
     // This number is calculated from K_MAX_NUM_BLOCKS to make
     // the 'the maximum value of the multiplier' close to 20.0
     uint256 public constant K_ZKPROOFS_PER_BLOCK = 1;
-    uint256 public constant K_VERIFICATION_DELAY = 60 minutes;
```
Comment (cyberhorsey) [inline packages/protocol/contracts/libs/LibConstants.sol]: why is this removed?

**Comment by cyberhorsey** [comment]
> 
We can do retryMessage in separate PR, to get this large opened one in when it is ready.

File: packages/protocol/test/bridge/Bridge.test.ts [hunk 2]
```diff
@@ -1010,11 +1014,675 @@ describe("integration:Bridge", function () {
...
+            const block: Block = await ethers.provider.send(
+                "eth_getBlockByNumber",
+                ["latest", false]
+            )
+
+            await headerSync.setSyncedHeader(block.hash)
+
+            const logsBloom = block.logsBloom.toString().substring(2)
+
+            const blockHeader: BlockHeader = {
```
Comment (cyberhorsey) [inline packages/protocol/test/bridge/Bridge.test.ts:1399]: we re-use this `get latest block and then convert logsbloom and then convert to blockheader` logic a bit, can it be in a utils function? Would remove a lot of lines from this file.

File: packages/protocol/test/bridge/Bridge.test.ts [hunk 3]
```diff
@@ -1010,11 +1014,675 @@ describe("integration:Bridge", function () {
...
+    })
+
+    describe("isMessageSent()", function () {
+        it("should return false, since no message was sent", async function () {
+            const { owner, l1Bridge, srcChainId, enabledDestChainId } =
+                await deployBridgeFixture()
+
+            const m: Message = {
+                id: 1,
+                sender: owner.address,
```
Comment (cyberhorsey) [inline packages/protocol/test/bridge/Bridge.test.ts:1032]: we also use this same message struct a lot, perhaps the `deployBridgeFixture` can declare this identical message and return it from there? Woudl remove a lot of lines from each individual test, where we just use same message.

**Review by cyberhorsey — DISMISSED** [review]

File: packages/protocol/contracts/libs/LibConstants.sol (hunk 1 above)
Comment (RogerLamTd) [reply packages/protocol/contracts/libs/LibConstants.sol]: I'm not sure, I think daniel updated it in an earlier commit

**Comment by RogerLamTd** [comment]
@cyberhorsey implemented suggested changes, also made the signalProof generation into a util function (similarly repeatable)

**Review by dantaik — DISMISSED** [review]

File: packages/protocol/tasks/utils.ts [hunk 4]
```diff
@@ -84,6 +86,67 @@ const MessageStatus = {
...
+async function getSignalProof(
+    hre: any,
+    contractAddress: string,
+    key: string,
+    blockHash: string,
+    blockHeader: BlockHeader
+) {
+    const proof: EthGetProofResponse = await hre.ethers.provider.send(
+        "eth_getProof",
+        [contractAddress, [key], blockHash]
```
Comment (davidtaikocha) [inline packages/protocol/tasks/utils.ts:128]: Seems the latest parameter should be an integer block number? 
Ref: https://eips.ethereum.org/EIPS/eip-1186

File: packages/protocol/contracts/libs/LibConstants.sol (hunk 1 above)
Comment (davidtaikocha) [reply packages/protocol/contracts/libs/LibConstants.sol]: This constant should have been removed in #380

Comment (RogerLamTd) [reply packages/protocol/contracts/libs/LibConstants.sol]: I'll mark this resolved then!

File: packages/protocol/tasks/utils.ts (hunk 4 above)
Comment (RogerLamTd) [reply packages/protocol/tasks/utils.ts:128]: Just switched it to block.number in the most recent push, please have a look!

File: packages/protocol/tasks/utils.ts [hunk 5]
```diff
@@ -120,7 +120,7 @@ async function getSignalProof(
     hre: any,
     contractAddress: string,
     key: string,
-    blockHash: string,
+    blockHash: number,
```
Comment (davidtaikocha) [inline packages/protocol/tasks/utils.ts:123]: Also rename to `blockNumber`?

**Review by davidtaikocha — APPROVED** [review]

**Review by dantaik — APPROVED** [review]
